samples/matter/bridge/src/bridge_manager.cpp

```diff
@@ -0,0 +1,392 @@
+		uint16_t clusterRevision = device->GetBridgedDeviceBasicInformationClusterRevision();
+		return HandleReadAttribute(reinterpret_cast<void *>(&clusterRevision), sizeof(clusterRevision), buffer,
+					   maxReadLength);
+	}
+	case Clusters::BridgedDeviceBasicInformation::Attributes::FeatureMap::Id: {
+		uint32_t featureMap = device->GetBridgedDeviceBasicInformationFeatureMap();
+		return HandleReadAttribute(reinterpret_cast<void *>(&featureMap), sizeof(featureMap), buffer,
+					   maxReadLength);
+	}
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+
+CHIP_ERROR BridgeManager::HandleReadDescriptor(BridgedDevice *device, AttributeId attributeId, uint8_t *buffer,
+					       uint16_t maxReadLength)
+{
+	switch (attributeId) {
+	case Clusters::Descriptor::Attributes::ClusterRevision::Id: {
+		uint16_t clusterRevision = device->GetDescriptorClusterRevision();
+		return HandleReadAttribute(reinterpret_cast<void *>(&clusterRevision), sizeof(clusterRevision), buffer,
+					   maxReadLength);
+	}
+	case Clusters::Descriptor::Attributes::FeatureMap::Id: {
+		uint32_t featureMap = device->GetDescriptorFeatureMap();
+		return HandleReadAttribute(reinterpret_cast<void *>(&featureMap), sizeof(featureMap), buffer,
+					   maxReadLength);
+	}
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+
+#if CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE
+CHIP_ERROR BridgeManager::HandleReadOnOffLight(OnOffLightDevice *device, AttributeId attributeId, uint8_t *buffer,
+					       uint16_t maxReadLength)
+{
+	switch (attributeId) {
+	case Clusters::OnOff::Attributes::OnOff::Id: {
+		bool onOff = device->GetOnOff();
+		return HandleReadAttribute(reinterpret_cast<void *>(&onOff), sizeof(onOff), buffer, maxReadLength);
+	}
+	case Clusters::OnOff::Attributes::ClusterRevision::Id: {
+		uint16_t clusterRevision = device->GetOnOffClusterRevision();
+		return HandleReadAttribute(reinterpret_cast<void *>(&clusterRevision), sizeof(clusterRevision), buffer,
+					   maxReadLength);
+	}
+	case Clusters::OnOff::Attributes::FeatureMap::Id: {
+		uint32_t featureMap = device->GetOnOffFeatureMap();
+		return HandleReadAttribute(reinterpret_cast<void *>(&featureMap), sizeof(featureMap), buffer,
+					   maxReadLength);
+	}
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+
+CHIP_ERROR BridgeManager::HandleWriteOnOffLight(OnOffLightDevice *device, AttributeId attributeId, uint8_t *buffer)
+{
+	switch (attributeId) {
+	case Clusters::OnOff::Attributes::OnOff::Id: {
+		device->Set(*buffer);
+		return CHIP_NO_ERROR;
+	}
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+}
+#endif /* CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE */
+
+#if CONFIG_BRIDGE_TEMPERATURE_SENSOR_BRIDGED_DEVICE
+CHIP_ERROR BridgeManager::HandleReadTemperatureSensor(TemperatureSensorDevice *device, AttributeId attributeId,
+						      uint8_t *buffer, uint16_t maxReadLength)
+{
+	switch (attributeId) {
+	case Clusters::TemperatureMeasurement::Attributes::MeasuredValue::Id: {
```

Comment:
Looks like we have many extensive switch cases with stateless calls in this implementation, maybe we could consider to reuse this simple map concept from WiFi bit:
https://github.com/nrfconnect/sdk-connectedhomeip/blob/master/src/platform/nrfconnect/wifi/WiFiManager.h#L47

Reply:
@markaj-nordic I managed to move the cluster and attribute ids to the specific devices implementations. Currently the only device types's dependency in bridge manager is the moment devices creation. Maybe I will figure out how to move it as well by the way of fixing other review comments or we will do the follow up to not block the other epic related tasks.